### DISCUSSION TITLE
Non standard Lab result format

### DISCUSSION THREAD

**Discussion by hartlin** [nextgenhealthcare/connect#5635] [General]
Using Mirth Connect 4.11

I am not a seasoned Javascript Programmer so I will need to explain why I need a true or false in an If else statement.

We are trying to accommodate lab results that is **NOT** in a standard format.

**I need to change this:**

OBX|2|ST|3141^HDL Cholesterol^L|1|1.10||>0.99|||A^S|F|||202301061738
OBX|3|ST|3150^Triglycerides^L|1|See Note||<1.71|||A^S|F|||202301061738
NTE|1||No result due to Bilirubin interference, suggest recollection.

**To This:**

OBX|2|ST|3141^HDL Cholesterol^L|1|1.10||>0.99|||A^S|F|||202301061738
OBX|3|ST|3150^Triglycerides^L|1|**See Note|**|<1.71|||A^S|F|||202301061738
NTE|1||3150^No result due to Bilirubin interference, suggest recollection.

This is my code, which works perfectly in this scenario but most OBX does not have a **"See Note"**
I need code that will check for a NTE after a OBX and return to a variable "true" so I can build the NTE to the remote labs specification.

var j = 0; j <= msg.elements('NTE').length()-1
for (var i = 0; i <= msg.elements('OBX').length()-1; i++) //Checks the next segment
{{		if (msg['OBX'][i]['OBX.5']['OBX.5.1'].toString() == "See Note")
		{msg['NTE'][j]['NTE.3']['NTE.3.1'] = msg['OBX'][i]['OBX.3']['OBX.3.1'].toString() +"^"+  msg['NTE'][j]['NTE.3']['NTE.3.1'] ;
		j++;
		}}
}

Been working on this for over a week and can't seem to find the right code.

**Comment by jonbartels** [comment]
Quick flyby -

1. Code formatting. Use the GH code formatting options and edit your post. It'll make the code easier to read
2. I think your conditional is fine
3. Can you show a FAILING example?
4. The essence of the problem is that you have an OBX and want to modify the NTE AFTER that OBX. OBX[n] is not necessarily paired with NTE[n] because there are not the same number of OBXs and NTEs
5. `getSegmentAfter` is kind of what you want, you have an OBX find the first NTE after that OBX. https://github.com/nextgenhealthcare/connect-examples/tree/master/Code%20Templates/Get%20Segments%20After%20a%20Particular%20Segment this example does exactly that
